lambda-buffers-frontend/app/LambdaBuffers/Frontend/Cli/Compile.hs

```diff
@@ -0,0 +1,25 @@
+module LambdaBuffers.Frontend.Cli.Compile (CompileOpts (..), compile) where
+
+import Control.Lens (makeLenses, (^.))
+import Data.Map qualified as Map
+import LambdaBuffers.Frontend.FrontM (runFrontM)
+import LambdaBuffers.Frontend.PPrint ()
+import Prettyprinter (Pretty (pretty))
+import Proto.Compiler ()
+
+data CompileOpts = CompileOpts
+  { _importPaths :: [FilePath]
+  , _moduleFilepath :: FilePath
+  }
+  deriving stock (Eq, Show)
+
+makeLenses ''CompileOpts
+
+compile :: CompileOpts -> IO ()
```

Comment:
We should add docs to all the exported functions of modules - I have to as well.

Reply:
Done, will apply throughout!